modules/ROOT/pages/traversal-framework/traversal_framework_java_api.adoc

```diff
@@ -0,0 +1,353 @@
+* `NODE_PATH` -- A node may not occur previously in the path reaching up to it.
+* `RELATIONSHIP_PATH` -- A relationship may not occur previously in the path reaching up to it.
+* `NODE_RECENT` -- Similar to `NODE_GLOBAL` uniqueness in that there is a global collection of visited nodes each position is checked against.
+This uniqueness level does however have a cap on how much memory it may consume in the form of a collection that only contains the most recently visited nodes.
+The size of this collection can be specified by providing a number as the second argument to the TraversalDescription.uniqueness()-method along with the uniqueness level.
+* `RELATIONSHIP_RECENT` -- Works like `NODE_RECENT` uniqueness, but with relationships instead of nodes.
+
+=== Example
+Here is an example of a traversal using a built-in `Uniqueness` constraint:
+----
+TraversalDescription td;
+try ( Transaction tx = graphDb.beginTx() ) {
+     td = tx.traversalDescription();
+            .uniqueness( Uniqueness.RELATIONSHIP_GLOBAL )
+}
+
+td.traverse( startNode );
+----
+
+For an example demonstrating how to use uniqueness in a traversal see <<examples-uniqueness-of-paths-in-traversals, Uniqueness of Paths in traversals>>.
+
+[[traversal-java-api-branchselector]]
+== BranchSelector
+
+A `BranchSelector` / `BranchOrderingPolicy` is used for selecting which branch of the traversal to attempt next.
+This is used for implementing traversal orderings.
+
+The Traversal Framework provides a few basic ordering implementations based on the link:https://en.wikipedia.org/wiki/Depth-first_search[depth-first^] and link:https://en.wikipedia.org/wiki/Breadth-first_search[breadth-first^] algorithms:
+
+* `BranchOrderingPolicies.PREORDER_DEPTH_FIRST` -- Traversing depth first, visiting each node before visiting its child nodes.
+* `BranchOrderingPolicies.POSTORDER_DEPTH_FIRST` -- Traversing depth first, visiting each node after visiting its child nodes.
+* `BranchOrderingPolicies.PREORDER_BREADTH_FIRST` -- Traversing breadth first, visiting each node before visiting its child nodes.
+* `BranchOrderingPolicies.POSTORDER_BREADTH_FIRST` -- Traversing breadth first, visiting each node after visiting its child nodes.
+
+[NOTE]
+====
+Breadth-first traversals have a higher memory overhead than depth-first traversals.
+====
+
+A `BranchSelector` carries state and hence needs to be uniquely instantiated for each traversal.
+Therefore it is supplied to the `TraversalDescription` through a `BranchOrderingPolicy` interface, which is a factory of `BranchSelector` instances.
+
+A user of the Traversal Framework rarely needs to implement their own `BranchSelector` or `BranchOrderingPolicy`, it is provided to let graph algorithm implementors provide their own traversal orders.
+The Neo4j Graph Algorithms package contains for example a `BestFirst` order `BranchSelector` / `BranchOrderingPolicy` that is used in BestFirst search algorithms such as A* and Dijkstra.
```

Comment:
Neo4j Graph Algorithms package? Do we mean GDS? Is the statement still true?

Reply:
https://neo4j.com/docs/java-reference/4.4/javadocs/org/neo4j/graphalgo/package-summary.html
https://neo4j.com/docs/java-reference/current/java-embedded/graph-algorithms/

It is a package and there are some algorithms as mentioned :) we have decided to add the link to the java docs to make it more clear though!